skimage/feature/texture.py

```diff
@@ -136,6 +140,13 @@ def graycomatrix(image, distances, angles, levels=None, symmetric=False, normed=
     if levels is None:
         levels = 256
 
+    if mask is not None:
+        if mask.shape != image.shape or mask.dtype != np.bool_:
+            raise ValueError(
+                "The mask must have the same shape as the image and be of boolean type."
+            )
+        image[~mask] = np.iinfo(image.dtype).max
```

Comment:
I think a comment explaining this bit would be really helpful to maintain and understand this code long-term. Do you want to give it a try?